.eslintrc.js

```diff
@@ -35,18 +35,44 @@ module.exports = {
       plugins: ['@typescript-eslint'],
       rules: {
         'default-case': 0,
-        'require-jsdoc': 'off',
         'no-undef': 'off',
         'react/prop-types': 'off',
         'no-shadow': 'off',
         '@typescript-eslint/no-empty-function': 'off',
         '@typescript-eslint/ban-ts-comment': 'off',
         'import/prefer-default-export': 'warn',
-        'valid-jsdoc': 'off',
         'newline-after-var': 'off',
         '@typescript-eslint/no-explicit-any': 'off',
         'lines-between-class-members': ['error', 'always', {exceptAfterSingleLine: true}],
         'tsdoc/syntax': 'off', // Todo: remove this before merging pr
+        'valid-jsdoc': [
+          'error',
+          {
+            prefer: {
+              arg: 'param',
+              argument: 'param',
+              return: 'returns',
+              virtual: 'abstract',
+              fires: 'emits',
+            },
+            requireParamDescription: false,
+            requireReturn: true,
+            requireReturnType: true,
```

Comment:
I don't think we need this in jsdoc as we have typescript now so you will have return type in the code

Reply:
ya but documentation needs it , we will need to work separately on the tsdoc and jsdoc migration